repose-aggregator/components/filters/valkyrie-authorization/src/test/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilterTest.scala

```diff
@@ -391,23 +391,272 @@ class ValkyrieAuthorizationFilterTest extends FunSpec with BeforeAndAfter with M
+
+      assert(mockServletResponse.getStatus == 502)
+    }
+
+    it("should 502 when valkyrie gives an unexpected response") {
+      setup
+      setMockAkkaBehavior(akkaServiceClient, "accounts", transformedTenant, contactId, 200, """{"banana":"phone"}""")
+
+      filter.doFilter(mockServletRequest, mockServletResponse, filterChain)
+
+      assert(mockServletResponse.getStatus == 502)
+    }
+
+    it("should 502 when we have an exception while talking to valkyrie") {
+      setup
+      Mockito.when(akkaServiceClient.get(Matchers.any(classOf[String]),
+        Matchers.eq(s"http://foo.com:8080/account/$transformedTenant/permissions/contacts/accounts/by_contact/$contactId/effective"),
+        Matchers.any(classOf[java.util.Map[String, String]]))).thenThrow(new AkkaServiceClientException("test exception", null))
+
+      filter.doFilter(mockServletRequest, mockServletResponse, filterChain)
+
+      assert(mockServletResponse.getStatus == 502)
+    }
+
+    it("should use the values from the datastore when available") {
+      setup
+      Mockito.when(mockDatastore.get(ValkyrieAuthorizationFilter.CACHE_PREFIX + "accounts" + transformedTenant + contactId))
+        .thenReturn(Vector("some_permission", "a_different_permission"), Nil:_*)
+      val captor = ArgumentCaptor.forClass(classOf[MutableHttpServletRequest])
+
+      filter.doFilter(mockServletRequest, mockServletResponse, filterChain)
+      Mockito.verify(filterChain).doFilter(captor.capture(), Matchers.any(classOf[ServletResponse]))
+
+      val roles: Iterator[String] = captor.getValue.getHeaders("X-Roles").asScala
+      assert(roles.contains("some_permission"))
+      assert(roles.contains("a_different_permission"))
+
+    }
+  }
+
+  describe("when permission to role translation and delegation is turned on") {
+    val config = createGenericValkyrieConfiguration(null)
+    config.setTranslatePermissionsToRoles(new Object)
+    config.setDelegating(new DelegatingType)
+    config.setCollectionResources(null)
+    val tenantId = "hybrid:98765"
+    val transformedTenant = "98765"
+    val contactId = "123456"
+    val akkaServiceClient = mock[AkkaServiceClient]
+    val filterChain = mock[FilterChain]
+    val mockServletRequest = new MockHttpServletRequest
+    val mockServletResponse = new MockHttpServletResponse
+    val filter = new ValkyrieAuthorizationFilter(mock[ConfigurationService], akkaServiceClient, mockDatastoreService)
+    filter.configurationUpdated(config)
+
+    def setup() = {
+      mockServletRequest.resetAll()
+      mockServletRequest.setMethod("GET")
+      mockServletRequest.setRequestURL("http://foo.com:8080")
+      mockServletRequest.setHeader("X-Tenant-Id", tenantId)
+      mockServletRequest.setHeader("X-Contact-Id", contactId)
+
+      mockServletResponse.resetAll()
+
+      Mockito.reset(filterChain)
+      Mockito.reset(akkaServiceClient)
+    }
+
+    it("should translate permissions to roles") {
+      setup
```

Comment:
More convention/Best Practice than standard. This is the document referenced by the Static Analysis tools in IntelliJ IDEA:
* http://www.cs.ucsb.edu/~benh/162_S15/scala/Programming-in-Scala.pdf
* 10.3 Defining parameterless methods

Which in turn references the *uniform access principle* in Meyer, Bertrand. Object-Oriented Software Construction. Prentice Hall, 2000.